### DISCUSSION TITLE
Disable AA on cursor:zoom_factor ?

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [hyprwm/Hyprland#6135] [Q&A] [answered]
is there a way to disable AntiAliasing on cursor zoom , just like  `hyprpicker` does ?
The zoom factor is not the same but the AA can be seen.

![image](https://github.com/hyprwm/Hyprland/assets/61395246/534fbfed-7cfa-4352-a8a7-d4789421a443)
![image](https://github.com/hyprwm/Hyprland/assets/61395246/71a751a5-f727-4d1d-8102-d96e37c8d6f1)

**Comment by vaxerski** [comment]
huh, I thought it'd be disabled by default, maybe it broke

**Reply by LamprosPitsillos** [reply]
Nah it's been like this ever since i started using Hyprland (last summer).
Should i move this to an issue?

**Reply by vaxerski** [reply]
prolly.

**Comment by BeyondMagic** [comment]
Since later issues were closed later and moved here, may I bump in the hopes this can be fixed.

The original poster search for the problem led him to:

> [Hyprland/src/render/Renderer.cpp](https://github.com/hyprwm/Hyprland/blob/f8857e6072bd85b95393499688872aaf7f088b5b/src/render/Renderer.cpp#L1257)
> 
> Line 1257 in [f8857e6](/hyprwm/Hyprland/commit/f8857e6072bd85b95393499688872aaf7f088b5b)
>  g_pHyprOpenGL->m_RenderData.useNearestNeighbor = false; 
> 
> I tried finding where the filtering was happening along with the cursor zoom and tried disabling with no luck.
> 
> Tried this too , even though this looks like a cleanup function, with no luck.
> 
> [Hyprland/src/render/OpenGL.cpp](https://github.com/hyprwm/Hyprland/blob/f8857e6072bd85b95393499688872aaf7f088b5b/src/render/OpenGL.cpp#L427)
> 
> Line 427 in [f8857e6](/hyprwm/Hyprland/commit/f8857e6072bd85b95393499688872aaf7f088b5b)
>  m_RenderData.useNearestNeighbor = true;

**Reply by ABrony** [reply]
Actually extremely annoying thing that even can't be worked around without poking into source code of hyprland (presumably). I was surprised when i first saw it. thought maybe there's different scaling/AA options in config, but nope

**Reply by BeyondMagic** [reply]
Yeah, unfortunately, not even tempering that part of the code make it works. Not sure what makes the antialiasing work yet.

**Reply by vaxerski** [reply]
on git thats no longer how aa is done so you might wanna check again? It's now passed as a param to the render fn. I'm travelling today so can't check.

**Reply by BeyondMagic** [reply]
i'll check it

**Reply by BeyondMagic** [reply, marked as answer]
This works @vaxerski, should I make it into a PR? I'm asking because you may prefer making it into an option and not force the no-anti-aliasing.

```diff
diff --git a/src/render/OpenGL.cpp b/src/render/OpenGL.cpp
index 3cb810e1..d0a3ba21 100644
--- a/src/render/OpenGL.cpp
+++ b/src/render/OpenGL.cpp
@@ -1760,6 +1760,17 @@ void CHyprOpenGLImpl::renderTexturePrimitive(SP<CTexture> tex, const CBox& box)
     glActiveTexture(GL_TEXTURE0);
     tex->bind();
 
+    // ensure the final blit uses the desired sampling filter
+    // when cursor zoom is active we want nearest-neighbor (no anti-aliasing)
+    if (m_renderData.useNearestNeighbor) {
+        tex->setTexParameter(GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+        tex->setTexParameter(GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+    } else {
+        tex->setTexParameter(GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+        tex->setTexParameter(GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+    }
+
     useProgram(shader->program);
     shader->setUniformMatrix3fv(SHADER_PROJ, 1, GL_TRUE, glMatrix.getMatrix());
     shader->setUniformInt(SHADER_TEX, 0);

```

**Reply by vaxerski** [reply]
yes this lgtm, make a MR
